### PR TITLE
Added CRUD for User and dummy dashboard page

### PR DESCRIPTION
**Pull request by drholmie**
Added basic user CRUD functionalities
Added email, username validation
Added basic dashboard page.
For #3 and partially #11 
Worked on with @danishebadulla326 

TODO
- Improve frontend
- Add posts, classes in dashboard

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,4 +3,9 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  root "users#index"
```
Comment (ySteinhart1) [inline config/routes.rb:6]: Small point, but if the user is logged in (```session[:current_user]``` is set), going to index should probably show something other than the login page

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            session[:current_user] = @user.id
+            redirect_to controller: 'users', action: 'show', id: @user.id 
+        else
+            message = "Something went wrong! Username or email incorrect!"
+            redirect_to login_path, alert: message
+        end
+      end
+
+      def new
+        @users = User.all
```
Comment (ySteinhart1) [inline app/controllers/users_controller.rb:18]: Is there a reason for needing @users here?

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        @user = User.new
+      end
+
+      def create
+        @user = User.new(user_params)
+        @users = User.all
```
Comment (ySteinhart1) [inline app/controllers/users_controller.rb:24]: Same comment as above here

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          render :new, status: :unprocessable_entity
+        end
+      end
+      
+      def edit
+        @user = User.find(params[:id])
+      end
+    
+      def update
+        @users = User.all
```
Comment (ySteinhart1) [inline app/controllers/users_controller.rb:39]: Same here

**Review by ySteinhart1 — CHANGES_REQUESTED** [review]

File: config/routes.rb (hunk 1 above)
Comment (ySteinhart1) [reply config/routes.rb:6]: I tried this and it seems to work:

`      def index
        if session[:current_user].nil?
            redirect_to login_path, alert: message
        else
            redirect_to controller: 'users', action: 'show', id: session[:current_user]
        end
      end`

File: app/controllers/users_controller.rb (hunk 4 above)
Comment (drholmie) [reply app/controllers/users_controller.rb]: yup changed this, thanks for catching it!

File: app/controllers/users_controller.rb (hunk 3 above)
Comment (drholmie) [reply app/controllers/users_controller.rb]: yup changed this, thanks for catching it!

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (drholmie) [reply app/controllers/users_controller.rb]: yup changed this, thanks for catching it!

File: config/routes.rb (hunk 1 above)
Comment (drholmie) [reply config/routes.rb:6]: Yeah changed this in the user_controller file as such : 
```
def index 
        if session[:current_user]
            redirect_to controller: 'users', action: 'show', id: session[:current_user]
        end
      end
     ```
. Tested it out and it works. Thanks for catching this!

**Review by ySteinhart1 — APPROVED** [review]

**Pull request merged**